### PR TITLE
fix: No context caching for json ld

### LINKED ISSUE #14: No context caching for json-ld documents
Description:

The current implementation doesn't cache context documents used in json-ld files. This causes tests and deployments to run into a rate limit and get banned for 24 hours.

### PR DESCRIPTION
**Pull request by borisrizov-zf**
## Description

This PR resolves #14 

## Pre-review checks

Please ensure to do as many of the following checks as possible, before asking for committer review:

- [x] DEPENDENCIES are up-to-date. [Dash license tool](https://github.com/eclipse/dash-licenses). Committers can open IP issues for restricted libs.
- [x] [Copyright and license header](https://eclipse-tractusx.github.io/docs/release/trg-7/trg-7-02) are present on all affected files

### CODE REVIEW

**Comment by borisrizov-zf** [comment]
@florianrusch-zf @DominikPinsel @koptan please review. This change is supposed to reflect the emergency update in the catenax-ng repo. Since the project structure is different I had to manually create the commits. I used the opportunity to order them logically and squash some of the commits from the original repo.

File: src/main/java/org/eclipse/tractusx/ssi/lib/proof/LinkedDataProofValidation.java [hunk 1]
```diff
@@ -46,20 +48,24 @@ public static LinkedDataProofValidation newInstance(DidResolver didResolver) {
...
   }
 
   private final LinkedDataHasher hasher;
   private final LinkedDataTransformer transformer;
   private final DidResolver didResolver;
+  private final JsonLdValidator jsonLdValidator;
 
   /**
-   * To verifiy {@link VerifiableCredential} or {@link VerifiablePresentation} In this method we are
+   * To verify {@link VerifiableCredential} or {@link VerifiablePresentation} In this method we are
```
Comment (florianrusch-zf) [inline src/main/java/org/eclipse/tractusx/ssi/lib/proof/LinkedDataProofValidation.java:63]:
```suggestion
   * To verify {@link VerifiableCredential} or {@link VerifiablePresentation}. In this method we are
```

**Review by florianrusch-zf — APPROVED** [review]
LGTM - just found a missing dot in a comment 😄

But I can only roughly assess whether these changes are OK or not - I don't know much about the whole SSI issue

**Review by florianrusch-zf — APPROVED** [review]

**Comment by borisrizov-zf** [comment]
@florianrusch-zf Have a look, I've rebased and included CHANGELOG instructions.

@DominikPinsel ping.

File: CHANGELOG.md [hunk 2]
```diff
@@ -5,8 +5,30 @@ All notable changes to this project will be documented in this file.
 The format is based on [Keep a Changelog](http://keepachangelog.com/en/1.0.0/)
 and this project adheres to [Semantic Versioning](http://semver.org/spec/v2.0.0.html).
 
+## [UNRELEASED]
+
+### Added
+
+- JSON-LD validation is now performed durin linked data validation.
```
Comment (florianrusch-zf) [inline CHANGELOG.md:12]:
```suggestion
- JSON-LD validation is now performed during linked data validation.
```

**Review by florianrusch-zf — APPROVED** [review]

**Comment by borisrizov-zf** [comment]
ping @DominikPinsel

**Comment by borisrizov-zf** [comment]
ping @koptan

**Review by DominikPinsel — APPROVED** [review]
Thank you :)

**Comment by koptan** [comment]
LGTM, thanks @borisrizov-zf  for creating this PR.

**Pull request merged**